platform/android/os_android.cpp

```diff
@@ -364,11 +366,59 @@ void OS_Android::process_joy_event(OS_Android::JoypadEvent p_event) {
 	}
 }
 
+void OS_Android::_set_key_modifier_state(Ref<InputEventWithModifiers> ev) const {
+	ev->set_shift(shift_mem);
+	ev->set_alt(alt_mem);
+	ev->set_metakey(meta_mem);
+	ev->set_control(control_mem);
+}
+
 void OS_Android::process_event(Ref<InputEvent> p_event) {
 
 	input->parse_input_event(p_event);
 }
 
+void OS_Android::process_key_event(int p_scancode, int p_unicode_char, bool p_pressed) {
+
+	Ref<InputEventKey> ev;
+	ev.instance();
+	int val = p_unicode_char;
+	int scancode = android_get_keysym(p_scancode);
+
+	if (scancode == KEY_SHIFT) {
```

Comment:
It would be cleaner to use `switch` or `else if` for these (no need to fix it on master as it's a minor change that can be done later).